cristin-import/src/main/java/no/unit/nva/cristin/utils/NvaCustomer.java

```diff
@@ -0,0 +1,135 @@
+                                                                   "bb3d0c0c-5065-4623-9b98-5810983c2478",
+                                                                   "api.dev.nva.aws.unit.no",
+                                                                   "bb3d0c0c-5065-4623-9b98-5810983c2478",
+                                                                   "api.test.nva.aws.unit.no",
+                                                                   "0baf8fcb-b18d-4c09-88bb-956b4f659103",
+                                                                   "api.e2e.nva.aws.unit.no",
+                                                                   "bb3d0c0c-5065-4623-9b98-5810983c2478",
+                                                                   "api.nva.unit.no",
+                                                                   "22139870-8d31-4df9-bc45-14eb68287c4a");
+    public static final String TOP = "top";
+    public static final String DEPTH = "depth";
+    public static final String CUSTOMER = "customer";
+    public static final String CRISTIN_ID = "cristinId";
+    public static final String PATH_DELIMITER = "/";
+    private URI id;
+    private URI cristinId;
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public Builder copy() {
+        return new Builder().withCristinId(this.cristinId).withId(this.id);
+    }
+
+    public static NvaCustomer fromCristinOrganization(URI cristinOrganizationId) {
+        return NvaCustomer.builder().withCristinId(cristinOrganizationId).build();
+    }
+
+    public NvaCustomer fetch(RawContentRetriever uriRetriever) {
+        return attempt(() -> uriRetriever.getRawContent(fetchCristinOrganizationUri(), APPLICATION_JSON))
+                   .map(Optional::orElseThrow)
+                   .map(NvaCustomer::toCristinOrganization)
+                   .map(CristinOrganization::getTopLevelOrganization)
+                   .map(CristinOrganization::id)
+                   .map(NvaCustomer::encode)
+                   .map(NvaCustomer::constructFetchNvaCustomerUri)
+                   .map(uri -> uriRetriever.getRawContent(uri, APPLICATION_JSON))
+                   .map(Optional::orElseThrow)
+                   .map(NvaCustomer::toNvaCustomerResponse)
+                   .map(NvaCustomerResponse::id)
+                   .map(id -> copy().withId(id).build())
+                   .orElse(failure -> emptyNvaCustomer());
+    }
+
+    private URI fetchCristinOrganizationUri() {
+        return UriWrapper.fromUri(cristinId).addQueryParameter(DEPTH, TOP).getUri();
+    }
+
+    private NvaCustomer emptyNvaCustomer() {
+        return new NvaCustomer();
+    }
+
+    private static NvaCustomerResponse toNvaCustomerResponse(String value) {
+        return attempt(() -> JsonUtils.dtoObjectMapper.readValue(value, NvaCustomerResponse.class)).orElseThrow();
+    }
```

Comment:
If I read this code correctly, the result of this query can be one of four outcomes. But you want to query 2 million cristin posts to an API in order to fetch that data? Sounds like an horrible idea to me.
And to top it all you want to handle the exception with orElseThrow?

Reply:
Snakket med Terje og han mener vi må ha riktig publisher for alle cristin poster, slik at personene ved alle de institusjonene får redigeringsmuligheter på sine importerte poster. Dersom vi cacher customer-api, så burde det vel gå greit?

Reply:
Men publisher blir satt til Sikt, det settes ikke til f.eks "NTNU". Vi har 4 miljøer. Dermed så får vi kun 4 muligheter. Og drive med IO operasjoner for dette virker unødvendig.

Reply:
orElseThrow() will end up in orElse(failure -> useSiktCustomer())

Reply:
Før du kan lage en PR, må du svare på spørsmålet hvem er det som er publisher på denne Cristin posten? Er det første forfatterens organisasjon? Er det personen som la inn posten i Cristin sin organisasjon.

Reply:
Og med du, så mener jeg at du må spørre Terje / Jan Erik.

Reply:
Akkurat nå satt jeg customer/publisher til tilhørende resourceOwner som settes på publikasjon.

Reply:
resourceOwner er hardkodet til å være Sikt?

Reply:
Ikke alltid ser det ut som
```
private ResourceOwner extractResourceOwner() {
        var cristinLocales = getValidCristinLocales();
        if (shouldUseOwnerCodeCreated(cristinLocales)) {
            return CristinLocale.builder()
                       .withOwnerCode(cristinObject.getOwnerCodeCreated())
                       .withInstitutionIdentifier(cristinObject.getInstitutionIdentifierCreated())
                       .withDepartmentIdentifier(cristinObject.getDepartmentIdentifierCreated())
                       .withSubDepartmentIdentifier(cristinObject.getSubDepartmendIdentifierCreated())
                       .withGroupIdentifier(cristinObject.getGroupIdentifierCreated())
                       .build().toResourceOwner();
        }
        if (cristinLocalesContainsCristinOwnerCodeCreated(cristinLocales)) {
            return bestMatchingResourceOwner(cristinLocales);
        }
        return Optional.of(cristinLocales)
                   .flatMap(list -> list.stream().findFirst())
                   .map(CristinLocale::toResourceOwner)
                   .orElse(SIKT_OWNER);
    }
```